Grant aspen batch jobs additional permissions

ttung opened the pull request:
### Description
1. This allows aspen batch jobs access to the secret that holds the gisaid credentials.
1. This allows aspen batch jobs access to the aspen-config secret.
1. This allows aspen batch jobs access to create more batch jobs.
1. This allows aspen batch jobs access to S3.

Depends on #202, #204

#### Issue
[ch126776](https://app.clubhouse.io/genepi/story/126776/workflow-to-download-and-process-raw-gisaid-data)

### Test plan
`make deploy-tf`

phoenixAja commented on terraform/aspen_batch_iam.tf:
```diff
@@ -1,6 +1,65 @@
+
+resource "aws_iam_policy" "aspen-batch-jobs-policies" {
+  name = "aspen-batch-jobs-policies"
+  policy = <<-POLICY
```
is this syntax correct or is it supposed to be `<<POLICY`? Why use this syntax instead of `<<EOF` (which seems to be what the terraform docs prefer)

ttung replied:
<<- allows you to indent the input.  see https://www.terraform.io/docs/language/expressions/strings.html#indented-heredocs

phoenixAja replied:
oh sweet didn't know that

phoenixAja commented on terraform/aspen_batch_iam.tf:
```diff
@@ -1,6 +1,65 @@
+                  "s3:PutObject"
+              ],
+              "Resource": [
+                  "arn:aws:s3:::${aws_s3_bucket.aspen-data.bucket}/raw_gisaid_dump/*",
+                  "arn:aws:s3:::${aws_s3_bucket.aspen-data.bucket}/processed_gisaid_dump/*",
+                  "arn:aws:s3:::${aws_s3_bucket.aspen-data.bucket}/aligned_gisaid_dump/*"
```
would be nice to move these key names to their own local variables eventually, just in case we need to use them in any other resource down the road.

ttung replied:
Yeah, I don't know if there's a way to share strings between terraform and python... @jgadling ?

Commit:
Bucket for data

### Description
This brings the aspen-data-dev bucket under terraform management.

### Test plan
```
% terraform import module.aspen.aws_s3_bucket.aspen-data aspen-data-dev
% make deploy-tf
```

Commit:
Grant aspen batch jobs additional permissions.

### Description
1. This allows aspen batch jobs access to the secret that holds the gisaid credentials.
1. This allows aspen batch jobs access to the aspen-config secret.
1. This allows aspen batch jobs access to create more batch jobs.

#### Issue
[ch126776](https://app.clubhouse.io/genepi/story/126776/workflow-to-download-and-process-raw-gisaid-data)

### Test plan
`make deploy-tf`
